tools/install_rust.sh

```diff
@@ -2,5 +2,5 @@
 if [ ! -d rust-installer ]; then
     mkdir rust-installer
     wget https://sh.rustup.rs -O rust-installer/rustup.sh
-    sh rust-installer/rustup.sh -y
+    sh rust-installer/rustup.sh -y --default-toolchain 1.73
```

Comment:
How much does this matter?  Won't Rustup just pull the right versions for everything that matters, since we're setting `RUSTUP_TOOLCHAIN`?

Reply:
It wouldn't matter much in practice as it should still work without this, I only did this to save an extra download.

Reply:
Removed in https://github.com/Qiskit/qiskit/pull/11267/commits/37a9a5219f7efbaa13ce57bdf8b1220c4b36d23f

Reply:
I was mostly thinking that this saves a download on macOS, but adds extra downloads on all other platforms?  Either way, not very important.

Reply:
Oh, this file only gets used on Linux builds. The script gets run inside the manylinux container (which doesn't include rust) as the first step. For macOS and windows we use the `rust-toolchain` action to install rust and we run everything directly in the ci vm directly.